source/gfoidl.Stochastics/Statistics/SampleBuilder.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using gfoidl.Stochastics.Builders;
+
+namespace gfoidl.Stochastics.Statistics
+{
+    public class SampleBuilder
+    {
+        // Must not be readonly, cf. https://gist.github.com/gfoidl/14b07dfe8ee5cb093f216f8a85759d88
+        private ArrayBuilder<double> _arrayBuilder;
+        private double _min = double.MaxValue;
+        private double _max = double.MinValue;
+        private double _sum;
+        //---------------------------------------------------------------------
+        public SampleBuilder() => _arrayBuilder = new ArrayBuilder<double>(true);
+        //---------------------------------------------------------------------
+        public void Add(double item)
+        {
+            _arrayBuilder.Add(item);
+
+            _sum += item;
+            if (item < _min) _min = item;
+            if (item > _max) _max = item;
+        }
+        //---------------------------------------------------------------------
+        public void Add(IEnumerable<double> values)
```

Comment:
Based on the results of https://github.com/gfoidl/Stochastics/pull/22#issuecomment-359278746 there should be "shortcuts" for `List<T>` and some other fast accessible enumerations.

Reply:
_Sample_ is immutable, once constructed. So this shortcut does not fit to _SampleBuilder_ , where it is possible to add further values at every time.

This means that even on the shortcut case the internal array (arraybuilder) must be updated to enable this.